components/tidb_query/src/rpn_expr/impl_json.rs

```diff
@@ -187,6 +187,34 @@ fn json_extract(args: &[ScalarValueRef]) -> Result<Option<Json>> {
     Ok(j.extract(&path_expr_list))
 }
 
+// Args should be like `(&Option<Json> , &[&Option<Bytes>])`.
+fn json_with_path_validator(expr: &tipb::Expr) -> Result<()> {
+    let children = expr.get_children();
+    assert!(children.len() == 1 || children.len() == 2);
```

Comment:
I think an error should returned instead. Invalid request from TiDB shouldn't make TiKV panic.

Reply:
I agree with you. But I find  that ```json_with_paths_validator``` contains ```assert!``` too. so I will fix it and open new PR to fix ```json_with_paths_validator```